pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -536,7 +524,7 @@ type EC2SDConfig struct {
 	// https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstances.html
 	// Filter API documentation: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Filter.html
 	// +optional
-	Filters []*EC2Filter `json:"filters"`
+	Filters []Filter `json:"filters"`
```

Comment:
This should be still a pointer as its a optional field

Reply:
Regarding the `[]RelabelConfig` field, since it is an array, I believe we can determine if the field was entered simply by checking its length. This approach should be sufficient, as even with pointers, we wouldn't be checking each element for nil.

PS: The change was originally suggested [here](https://github.com/prometheus-operator/prometheus-operator/pull/6633#discussion_r1620243723).

Reply:
I now see that `omitempty` is not present here. Should I add that?

Reply:
yes please add the omitempty tag.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required says:

````
Optional fields have the following properties:
[...]
* [...] have a built-in nil value (e.g. maps and slices).
````